src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -119,22 +119,21 @@
   (atom {}))
 
 (defonce ^:private ^{:doc "A map of DB details hash values, keyed by Database `:id`."}
-  database-id->db-details-hashes
+  database-id->jdbc-spec-hash
   (atom {}))
 
-(defn- db-details-hash
-  "Computes a hash value for the given `database`'s `:details` map, for the purpose of determining if details changed
-  and therefore the existing connection pool needs to be invalidated."
-  [database]
+(defn- jdbc-spec-hash
+  "Computes a hash value for the JDBC connection spec based on `database`'s `:details` map, for the purpose of
+  determining if details changed and therefore the existing connection pool needs to be invalidated."
+  [{driver :engine, :keys [details], :as database}]
   {:pre [(or nil? (instance? (type Database) database))]}
-  (if (some? database)
-    (hash (:details database))
-    nil))
+  (when (some? database)
+    (hash (connection-details->spec driver details))))
```

Comment:
Just hashing `details` isn't enough because some databases such as Snowflake will change the actual underlying connection parameters based on other factors such as the `start-of-week` Setting. Now we hash the JDBC spec and use that as a key instead